pele_platform/analysis/analysis.py

```diff
@@ -387,76 +420,112 @@ def generate_clusters(self, path, clustering_type,
+        )
         rmsd_per_cluster = self._calculate_cluster_rmsds(clusters, coordinates)
 
-        cluster_summary = self._analyze_clusters(clusters, dataframe,
-                                                 rmsd_per_cluster)
+        cluster_summary = self._analyze_clusters(clusters, dataframe, rmsd_per_cluster)
 
         if len(cluster_summary) == 0:
-            print(f"No clusters could be obtained, " +
-                  f"clustering analysis is skipped")
+            print(
+                f"No clusters could be obtained, " + f"clustering analysis is skipped"
+            )
 
             return
 
-        cluster_subset, cluster_summary = \
-            self._select_top_clusters(clusters, cluster_summary,
-                                      top_clusters_criterion,
-                                      max_clusters_to_select=max_top_clusters,
-                                      min_population_to_select=min_population)
+        cluster_subset, cluster_summary = self._select_top_clusters(
+            clusters,
+            cluster_summary,
+            top_clusters_criterion,
+            max_clusters_to_select=max_top_clusters,
+            min_population_to_select=min_population,
+        )
 
         # If water coordinates have been extracted, use them to locate
         # main water sites for each top cluster
         if water_coordinates is not None:
             self._get_water_sites(cluster_subset, water_coordinates, path)
 
-        print(f"Retrieve top clusters based on " +
-              f"{metric_top_clusters_criterion[top_clusters_criterion]}.")
+        print(
+            f"Retrieve top clusters based on "
+            + f"{metric_top_clusters_criterion[top_clusters_criterion]}."
+        )
 
         # Save cluster summary to file with information about selected labels
         cluster_summary.to_csv(os.path.join(path, "info.csv"), index=False)
 
         # Plot cluster descriptors
-        self._plot_cluster_descriptors(cluster_subset, dataframe,
-                                       cluster_summary, path)
+        self._plot_cluster_descriptors(cluster_subset, dataframe, cluster_summary, path)
 
         # Plot clusters
         self._plot_clusters(cluster_subset, dataframe, cluster_summary, path)
 
-        # Save cluster representative structures
-        self._save_cluster_representatives(cluster_subset, dataframe, path,
-                                           representatives_criterion)
+        # Save cluster representative structures or run inner clustering
+        if "multi" in representatives_criterion.lower():
+            self._run_inner_clustering(
+                path=path,
+                selected_clusters=cluster_subset,
+                all_clusters=clusters,
+                coordinates=coordinates,
+                representatives_criterion=representatives_criterion,
+                dataframe=dataframe,
+            )
```

Comment:
we can do:
```python
criterion_message = "k-means clustering"
```

right inside this conditional and forget about the next try except checking